bin/dataduct

```diff
@@ -126,57 +116,99 @@ class _HelpAction(argparse._HelpAction):
             for choice, subparser in subparsers_action.choices.items():
                 print "Command '{}'".format(choice)
                 print subparser.format_usage()
-
         parser.exit()
 
 
+def choice_help_formatter(choices):
+    """Create help format for a choice argument
+    """
+    result = []
+    for key, value in choices.iteritems():
+        result.append('%s: %s' % (key, value))
+    return '\n'.join(result)
+
+
 def main():
-    """Main function"""
-    parser = argparse.ArgumentParser(description='Run Dataduct commands',
-                                     add_help=False)
-    parser.add_argument(
+    """Main function
+    """
+
+    # Overwrite default help
+    help_parser = argparse.ArgumentParser(
+        description='Help message',
+        add_help=False)
+    help_parser.add_argument(
+        '-h',
+        '--help',
+        action=_HelpAction,
+        help='Help message',
+    )
+
+    # Base parser for all commands
+    base_parser = argparse.ArgumentParser(
+        description='Base Parser',
+        add_help=False,
+    )
+    base_parser.add_argument(
+        '-F',
+        '--filename',
+        default=None,
+        help='Filename to store output of commands',
+    )
+    base_parser.add_argument(
         '-m',
         '--mode',
         default=None,
         help='Mode to run the pipeline and config overrides to use',
     )
-    # Overwrite default help
-    parser.add_argument(
-        '-h',
-        '--help',
-        action=_HelpAction,
-        help='Show this help message and exit',
+
+    # Main parser
+    parser = argparse.ArgumentParser(
```

Comment:
Can you move this parser above `base_parser`? It's really unclear because `base_parser` is the only the base parser for the subparsers, not actual parser itself.